R/propose_snapshot.R

```diff
@@ -0,0 +1,55 @@
+#' @title Propose snapshot
+#' @export
+#' @family staging
+#' @description Propose a Production snapshot of Staging.
+#' @details [recommend_snapshot()] proposes a snapshot of Staging
+#'   to migrate to Production. The recommended snapshot is the list of
+#'   packages for which (1) the build and check results of the current
+#'   release are in Staging, and (2) there are no issues.
+#' @return `NULL` (invisibly)
+#' @inheritParams update_staging
+#' @param repo_staging Character string, URL of the staging universe.
+#' @examples
+#' \dontrun{
+#' url_staging = "https://github.com/r-multiverse/staging"
+#' path_staging <- tempfile()
+#' gert::git_clone(url = url_staging, path = path_staging)
+#' propose_snapshot(
+#'   path_staging = path_staging,
+#'   repo_staging = "https://staging.r-multiverse.org"
+#' )
+#' }
+propose_snapshot <- function(
```

Comment:
propose `prepare_snapshot()`?

Reply:
I thought "propose" felt better than "prepare" because the latter makes it sound like we might actually be creating the real snapshot and downloading all the packages. "propose" feels like the weaker condition, which seems right since it will be running every time Staging is updated.

Reply:
if it's designed to run each time then that makes sense.